exchanges/poloniex/poloniex_types.go

```diff
@@ -1,385 +1,73 @@
-	Trades      map[string][]ResultingTrades `json:"resultingTrades"`
-}
-
-// Withdraw holds withdraw information
-type Withdraw struct {
-	Response string `json:"response"`
-	Error    string `json:"error"`
-}
-
-// Fee holds fees for specific trades
-type Fee struct {
-	MakerFee        float64 `json:"makerFee,string"`
-	TakerFee        float64 `json:"takerFee,string"`
-	ThirtyDayVolume float64 `json:"thirtyDayVolume,string"`
-}
-
-// Margin holds margin information
-type Margin struct {
-	TotalValue    float64 `json:"totalValue,string"`
-	ProfitLoss    float64 `json:"pl,string"`
-	LendingFees   float64 `json:"lendingFees,string"`
-	NetValue      float64 `json:"netValue,string"`
-	BorrowedValue float64 `json:"totalBorrowedValue,string"`
-	CurrentMargin float64 `json:"currentMargin,string"`
-}
-
-// MarginPosition holds margin positional information
-type MarginPosition struct {
-	Amount           float64 `json:"amount,string"`
-	Total            float64 `json:"total,string"`
-	BasePrice        float64 `json:"basePrice,string"`
-	LiquidationPrice float64 `json:"liquidationPrice"`
-	ProfitLoss       float64 `json:"pl,string"`
-	LendingFees      float64 `json:"lendingFees,string"`
-	Type             string  `json:"type"`
-}
-
-// LoanOffer holds loan offer information
-type LoanOffer struct {
-	ID        int64   `json:"id"`
-	Rate      float64 `json:"rate,string"`
-	Amount    float64 `json:"amount,string"`
-	Duration  int64   `json:"duration"`
-	AutoRenew bool    `json:"autoRenew"`
-	Date      string  `json:"date"`
-}
-
-// ActiveLoans shows the full active loans on the exchange
-type ActiveLoans struct {
-	Provided []LoanOffer `json:"provided"`
-	Used     []LoanOffer `json:"used"`
-}
-
-// LendingHistory holds the full lending history data
-type LendingHistory struct {
-	ID       int64   `json:"id"`
-	Currency string  `json:"currency"`
-	Rate     float64 `json:"rate,string"`
-	Amount   float64 `json:"amount,string"`
-	Duration float64 `json:"duration,string"`
-	Interest float64 `json:"interest,string"`
-	Fee      float64 `json:"fee,string"`
-	Earned   float64 `json:"earned,string"`
-	Open     string  `json:"open"`
-	Close    string  `json:"close"`
-}
-
-// WebsocketTicker holds ticker data for the websocket
-type WebsocketTicker struct {
-	CurrencyPair  string
-	Last          float64
-	LowestAsk     float64
-	HighestBid    float64
-	PercentChange float64
-	BaseVolume    float64
-	QuoteVolume   float64
-	IsFrozen      bool
-	High          float64
-	Low           float64
-}
-
-// WebsocketTrollboxMessage holds trollbox messages and information for
-// websocket
-type WebsocketTrollboxMessage struct {
-	MessageNumber float64
-	Username      string
-	Message       string
-	Reputation    float64
-}
-
-type wsOp string
-
-const (
-	wsSubscribeOp   wsOp = "subscribe"
-	wsUnsubscribeOp wsOp = "unsubscribe"
+type (
+	timeInForce order.TimeInForce
+	orderType   order.Type
+	accountType asset.Item
```

Comment:
🚧 Change request:
Pretty sure I saw someone else mention this too, but I can't see it now.
These need to be public if they're accepted or used in poloniex.go 🙂